src/Uno.Extensions.Reactive.UI/Presentation/Bindings/Collections/NullObservableVector.cs

```diff
@@ -0,0 +1,75 @@
+﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Linq;
+using Windows.Foundation.Collections;
+
+namespace Uno.Extensions.Reactive.Bindings.Collections
+{
+	/// <summary>
+	/// Null pattern implementation of <see cref="IObservableVector{T}"/>.
+	/// </summary>
+	internal class NullObservableVector<T> : IObservableVector<T>
+	{
+#pragma warning disable CS0067 // Event not raised
+		/// <inhertidoc />
+		public event VectorChangedEventHandler<T>? VectorChanged;
+#pragma warning restore CS0067
+
+		/// <inhertidoc />
+		public int Count { get; } = 0;
+
+		/// <inhertidoc />
+		public bool IsReadOnly { get; } = true;
+
+		/// <inhertidoc />
+		public T this[int index]
+		{
+			get => default!;
+			set => throw new NotSupportedException();
+		}
+
+		/// <inhertidoc />
+		public IEnumerator<T> GetEnumerator() => new Enumerator();
+
+		/// <inhertidoc />
+		IEnumerator IEnumerable.GetEnumerator() => new Enumerator();
+
+		/// <inhertidoc />
+		public void Add(T item) => throw new NotSupportedException();
+		
+		/// <inhertidoc />
+		public void Clear() => throw new NotSupportedException();
+
+		/// <inhertidoc />
+		public bool Contains(T item) => false;
```

Comment:
Is there a reason that Clear throws an exception whereas Contains returns false? Shouldn't Contains throw a NotSupportedException too? Same for CopyTo, Remove etc

Reply:
This collection is flagged as `ReadOnly` so we throw exception in all methods that attempt to modify the collection (i.e. write), while we do allow "read" methods.